File: anika.dovob/src/main/java/com/example/anika/homework/chapter7/Item.java
package com.example.anika.homework.chapter7;

import jakarta.persistence.*;

@Entity(name = "Ch7Item")
@Table(name = "item_ch7")
@Inheritance(strategy = InheritanceType.JOINED)
@DiscriminatorColumn(name = "dtype")
public abstract class Item extends BaseEntity {

    @Id
    @GeneratedValue
    @Column(name = "item_id")
    private Long id;

    @Column(nullable = false)
    private String name;

    @Column(name = "base_price", nullable = false)
    private long basePrice;

    protected Item() {
    }

    public Item(String name, long basePrice) {
        this.name = name;
        this.basePrice = basePrice;
    }

    public Long getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public long getBasePrice() {
        return basePrice;
    }
}

File: hyun.k/hyun.k/src/main/kotlin/chapter7/Item.kt
package chapter7

import jakarta.persistence.DiscriminatorColumn
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.Id
import jakarta.persistence.Inheritance
import jakarta.persistence.InheritanceType

@Entity
@Inheritance(strategy = InheritanceType.JOINED)
@DiscriminatorColumn(name = "DTYPE")
abstract class Item : BaseEntity() {
    @Id
    @GeneratedValue
    var id: Long? = null

    var name: String? = null
    var basePrice: Long = 0

}